core/includes/updater.inc

```diff
@@ -175,19 +175,23 @@ abstract class Updater implements BackdropUpdaterInterface {
   }
 
   /**
-   * Return the project type from a Backdrop info file.
+   * Return the project type from a Backdrop info file or directory.
    *
    * @param string $directory
-   *   Directory to search for the info file.
+   *   Stream wrapper to directory to get the type from.
    *
    * @return string
-   *   The type of the project, either "module", "theme", or "layout".
+   *   The type of the project, either "core", "module", "theme", or "layout".
    *
    * @throws UpdaterException
    */
   public static function getProjectType($directory) {
-    if ($directory == BACKDROP_ROOT) {
-      return 'core';
+    $basename = basename($directory);
+    if (substr($basename, 0, 8) == 'backdrop') {
```

Comment:
...this may match things like `backdrop_cool_module` or `backdrop_neat_theme`. I think that it would be better to use `preg_match`:

```suggestion
    if (substr($basename, 0, 8) == 'backdrop') {
    if preg_match('/^backdrop(-[\d\.]+)?/', $basename) {
```

untested ^^

Reply:
Agreed 👍

Reply:
Good catch @herbdool 👍 ...were you thinking a `continue` perhaps?

Reply:
We found a consensus, that it's either exactly "backdrop", or it can't be backdrop core.
Regarding @herbdool's remark, that if the check for "core" fails, nothing happens. That's not the case. An UpdaterExeption is thrown by getUpdaterFromDirectory(), as getProjectType() returns nothing then.